### ISSUE TITLE
Show a players ranks within the clan

### ISSUE THREAD

**Issue by Stickaplex** [wise-old-man/wise-old-man#1296]
Currently for the categories it only shows the order of players for that particular category(eg. hitpoints, hydra kc, total clues). Would be great if you could select a player and get the rank of all of various categories.

Not sure the best way to implement this tbh. Perhaps on the individual players page, next to the category selection dropdown, have a different dropdown menu that can select the different groups or the global/ironman highscores. This way can even do gained/lost rank as well. Might not be the best way, but just what I came up with as I've been writing this suggestion.

**Comment by Sharpienero** [comment]
Hi @Stickaplex 

I am looking for some clarification on this.

If you currently hit the route (https://api.wiseoldman.net/v2/groups/5887/hiscores?metric=agility&limit=5), it returns a sorted array by overall rank. What you're proposing is that this route also returns a new field called something along the lines of clanRank that simply returns the index they are in the array?

I may be wrong, but this seems more like an implementation problem than an API problem. Would love other's input here.

**Comment by psikoi** [comment]
I think this feature is unfeasible to implement,

1) It's ambiguous what a player's "clan" is as they might be in different groups at the same time
2) To calculate a player's group rank in a skill, we'd have to pull everyone's stats and then sort to find their position

**Comment by Stickaplex** [comment]
@psikoi for point 1 i was thinking it would just be a drop down menu similiar to how it selects whats data to show (skills, bosses, clues etc.) would assume it could be just to the left of this current dropdown menu https://i.imgur.com/h77tYU4.png figured it could show things like overall rank, group 1, group 2, ironman rank, etc

as for point 2, im not too technical of a person so if its too heavy on the system i cant really argue that

**Comment by Sharpienero** [comment]
So you were thinking something like this?

![image](https://github.com/wise-old-man/wise-old-man/assets/8890281/e8640bff-5267-473c-8d49-777babd14584)

I thought it was an API modification but I see what you mean on the front-end. Why not just go to the group affiliation and find the user in their group ranking?

**Comment by Stickaplex** [comment]
@Sharpienero yes thats basically what i was talking about. but preferably 2 separate drop downs. one for skills/bosses and the other for comparing what groups/overall to rank you.

and because as far as im aware it would only show the rank for that particular stat. not to mention in larger groups it can be a bit more annoying to find a player, especially if they arent at the top of the leaderboard
